Code/Mantid/Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -644,6 +679,39 @@ MatrixWorkspace_sptr ReflectometryReductionOne::transmissonCorrection(
 }
 
 /**
+ * Perform Transmission Corrections Using Polynomials.
+ * @param IvsLam : Run workspace which is to be normalized by the results of the
+ * transmission corrections.
+ * @return Normalized run workspace by the generated polynomial.
+ */
+MatrixWorkspace_sptr
+ReflectometryReductionOne::polynomialCorrection(MatrixWorkspace_sptr IvsLam) {
+
+  const std::string corrAlgName = getProperty("CorrectionAlgorithm");
+
+  const std::string polynomial = getProperty("Polynomial");
+  const std::string c0 = getProperty("C0");
+  const std::string c1 = getProperty("C1");
+
+  IAlgorithm_sptr corrAlg = createChildAlgorithm(corrAlgName);
+  corrAlg->initialize();
+  if (corrAlgName == "PolynomialCorrection") {
+    corrAlg->setProperty("Coefficients", polynomial);
+  } else if (corrAlgName == "ExponentialCorrection") {
+    corrAlg->setProperty("C0", c0);
+    corrAlg->setProperty("C1", c1);
+  } else {
```

Comment:
I think variables ```polynomial```, ```c0``` and ```c1``` should be declared in the clauses where they are used.